src/libmongoc/src/mongoc/mongoc-database.c

```diff
@@ -1029,3 +1032,34 @@ mongoc_database_watch (const mongoc_database_t *db,
 {
    return _mongoc_change_stream_new_from_database (db, pipeline, opts);
 }
+
+bool
+mongoc_database_set_timeout_ms (mongoc_database_t *db,
+                                int64_t timeout_ms,
+                                bson_error_t *error)
+{
+   BSON_ASSERT_PARAM (db);
+
+   if (timeout_ms < -1) {
```

Comment:
Same comment, I think this can be a strict negative check and disallow `-1` too.

Reply:
Will change.